src/test/java/seedu/address/logic/commands/ClientNoteEditCommandTest.java

```diff
@@ -0,0 +1,124 @@
+        ClientNoteEditCommand clientNoteEditCommand = new ClientNoteEditCommand(clientIdx, clientNoteIdx, newEditNote);
+        assertCommandSuccess(clientNoteEditCommand, newModel, expectedResult, expectedModel);
+    }
+
+    @Test
+    public void execute_validClientIdxValidNoteIdxNewTaggedNote_preservesTagHistory() {
+        Tag oldTag = new Tag("historicalTag");
+        Tag newTag = new Tag("newTag");
+        Set<Tag> newTagSet = new HashSet<>();
+        newTagSet.add(newTag);
+        Set<Tag> tagHistory = new HashSet<>();
+        tagHistory.add(oldTag);
+        Set<Tag> expectedTags = new HashSet<>(); // preserves old tags and new tags
+        expectedTags.add(oldTag);
+        expectedTags.add(newTag);
+        Index clientIdx = Index.fromOneBased(1);
+        Index clientNoteIdx = Index.fromOneBased(1);
+        Note oldClientNote = new Note(NOTE_CONTENT_1);
+        oldClientNote.setTags(tagHistory);
+        Note clientNote2 = new Note(NOTE_CONTENT_2);
+        Note newNote = new Note("dummy note to edit previous note");
+        newNote.setTags(newTagSet);
+        Model newModel = new ModelManager();
+        Client client1 = new ClientBuilder().withName("client1").build();
+        newModel.addClient(client1);
+        newModel.addClientNote(client1, oldClientNote);
+        newModel.addClientNote(client1, clientNote2);
+
+        Model expectedModel = new ModelManager();
+        Client client1Copy = new ClientBuilder().withName("client1").build();
+        expectedModel.addClient(client1Copy);
+        newNote.setTags(expectedTags);
+        expectedModel.addClientNote(client1Copy, newNote);
+        expectedModel.addClientNote(client1Copy, clientNote2);
+
+        CommandResult expectedResult = new CommandResult(ClientNoteEditCommand.MESSAGE_EDIT_CLIENT_NOTE_SUCCESS);
+        ClientNoteEditCommand clientNoteEditCommand = new ClientNoteEditCommand(clientIdx, clientNoteIdx, newNote);
+        assertCommandSuccess(clientNoteEditCommand, newModel, expectedResult, expectedModel);
+    }
+
+    @Test
+    public void testEquals() {
+        Note testNote = new Note("need for speed");
+        ClientNoteEditCommand clientNoteEditCommand1 = new ClientNoteEditCommand(Index.fromOneBased(1),
+                Index.fromOneBased(1), testNote);
+        ClientNoteEditCommand clientNoteEditCommand1Duplicate = new ClientNoteEditCommand(Index.fromOneBased(1),
+                Index.fromOneBased(1), testNote);
+        ClientNoteEditCommand clientNoteEditCommandClient2 = new ClientNoteEditCommand(Index.fromOneBased(2),
+                Index.fromOneBased(1), testNote);
+        ClientNoteEditCommand clientNoteEditCommand1ClientNote2 = new ClientNoteEditCommand(Index.fromOneBased(1),
+                Index.fromOneBased(2), testNote);
+        Object randomObject = new Object();
+
+        // random object -> returns false
+        assertFalse(clientNoteEditCommand1.equals(randomObject));
+
+        // same object -> returns true
```

Comment:
I think `same object` refers to comparing with itself which leads to the short circuit clause right? In this case, it's comparing with another object that has the same fields.

Reply:
Resolved